waspc/cli/src/Wasp/Cli/Command/Telemetry/User.hs

```diff
@@ -2,19 +2,28 @@
 
 module Wasp.Cli.Command.Telemetry.User
   ( UserSignature (..),
-    readOrCreateUserSignatureFile,
+    obtainUserSignature,
   )
 where
 
 import qualified Data.UUID.V4 as UUID
 import StrongPath (Abs, Dir, File', Path', relfile)
 import qualified StrongPath as SP
 import qualified System.Directory as SD
+import qualified System.Environment as ENV
 import Wasp.Cli.Command.Telemetry.Common (TelemetryCacheDir)
+import Wasp.Util (checksumFromString, hexToString)
 
 -- Random, non-identifyable UUID used to represent user in analytics.
 newtype UserSignature = UserSignature {_userSignatureValue :: String} deriving (Show)
 
+obtainUserSignature :: Path' Abs (Dir TelemetryCacheDir) -> IO UserSignature
+obtainUserSignature telemetryCacheDirPath = do
+  envSign <- getUserSignatureFromEnv
```

Comment:
Very minor, but I'd probably shorten to `envSig`.

Reply:
Good idea, will do! `sig` is more clearly `signature` than `sign` is!

Reply:
I actually ended up removing this name by golfing the code somewhat, which I believe resulted in more readable and nicer code! Give it a look.

Reply:
Btw looking at it a bit more, I probably should have implemented "monadic" version of `fromMaybe`, as:
```hs
fromMaybeM :: (Monad m) => m a -> m (Maybe a) -> m a
fromMaybeM ma = (>>= maybe ma return)
```

Reminder: signature of `fromMaybe` is:
```hs
fromMaybe :: a -> Maybe a -> a
```

But I do love the verbosity of `butIfNothingDo` :D! Even if it is just `fromMaybeM` with flipped parameters.

Reply:
Ok I did both! I implemented `fromMaybeM`, but then also `orIfNothingM` and `orIfNothing` because I think that is much more readable than `fromMaybe` or `fromMaybeM`.

Reply:
Ah  fancy 🕺🏻 , I like it :D played a bit with it in the REPL for my meager mind to grok but 👍🏻

Reply:
lol @shayneczyzewski :D. I didn't just come up with it from nothing, I golfed and golfed till it became this :D. My feeble mind just had more time to play with it.